Clean up .map files if they exist

ckornell opened the pull request:
In dev mode, I kept getting a .map file, and deleting the file again & again got to be a tab cumbersome. Let me know if you want this handled differently; I would love not to use a fork for something this simple. Thanks for this awesome plugin!

cascornelissen commented on lib/index.js:
```diff
@@ -276,17 +276,25 @@ module.exports = class SVGSpritemapPlugin {
             }
 
             compilation.chunks.forEach((chunk) => {
+                const sourceMapStr = `${chunk.files[0]}.map`;
```
Is the sourcemap `.map` file not included in `chunk.files`? 🤔

ckornell replied:
The sourcemap files are in `chunk.files`. Sorry! I'm not sure I understand the question. Can you clarify?

ckornell replied:
I think I got you mean. The sourcemap file, if one exists, is at the 2nd index (`chunk.files[1]`) I did the string template so we could ensure we're always just cleaning up the sourcemap file for the pertinent chunk. I could do a check for the `.map` extension on the 2nd index and delete it if it's true. Would that be clearer?

cascornelissen replied:
We can just do the following on line 294 (I think, might need some testing) as this will only delete the index when the chunk name matches (which is checked on line 289).

```es6
if ( typeof chunk.files[1] !== 'undefined' ) {
    delete assets[chunk.files[1]];
}
```

ckornell replied:
Sounds good. I'll test that now.

ckornell replied:
Tested and pushed.

cascornelissen commented on lib/index.js:
```diff
@@ -276,17 +276,25 @@ module.exports = class SVGSpritemapPlugin {
                 }
 
-                delete compilation.assets[chunk.files[0]];
+                delete assets[chunk.files[0]];
```
This should be `compilation.assets` instead of just `assets`, right?

ckornell replied:
No, I defined `assets` on line [279](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/pull/79/files#diff-6d186b954a58d5bb740f73d84fe39073R279) so we didn't have to repeat `.compilation`

cascornelissen replied:
Let's keep it at `compilation.assets`, it's a little more verbose which I like because we're making changes to the `compilation` object from the surrounding scope.

ckornell commented on lib/index.js:
```diff
@@ -276,17 +276,24 @@ module.exports = class SVGSpritemapPlugin {
                 }
 
-                delete compilation.assets[chunk.files[0]];
+                if ( typeof chunk.files[1] !== 'undefined' && chunk.files[1].indexOf('.map') !== -1 ) {
```
`chuck.files[1]` can be the actual .svg output, so need to make sure it's a sourcemap file at this index before removing.

cascornelissen replied:
🤔 In that case it might be better to just check for the entire filename as `[1]` might as well be a `.map` file for another file...

cascornelissen commented:
@ckornell, made some more minor comments. If you can add a test I'll look into the comments myself, of course you're free to fix them as well ✌️

ckornell commented:
Hey @cascornelissen, I made the changes you requested and pushed up the commit. I attempted to write a test, but I'm afraid I'm too new to webpack plugin development to figure it out. Here's what I tried...
``` 
it('Deletes sourcemap files if it exists', async () => {

    webpack({
        entry: './webpack/index.js',
        mode: 'development',
        plugins: [
            new SVGSpritemapPlugin({
                output: {
                    filename: path.join(__dirname, 'output/svg/test.svg')
                }
            })
        ]
    });

    const outputSourceMap = fs.existsSync(path.join(__dirname, 'output/svg/test.map'));

    expect(outputSourceMap).toBeFalsy();
}) 
```

Probably need to wait for webpack but am not sure how to do that here. Sorry!

cascornelissen commented:
LGTM 👍 Don't worry about the test too much, I can finish that part up as well this weekend when I hope to merge and release this change.

There's some tests that kick off webpack here if you still want to try 😉 
https://github.com/cascornelissen/svg-spritemap-webpack-plugin/blob/master/tests/svg4everybody-helper.test.js

ckornell commented:
Hey @cascornelissen, I added some working tests. I'm sure they could be done better. I figured since we're testing the sourcemap file we should do the same for the JS file, so I added that too. Thanks for all your help! Let me know if I can do anything else. Have a great weekend.

cascornelissen commented:
@ckornell, thanks one last time for your contribution! ✌️ I've cleaned some stuff up (in 701c3e5 if you're interested) and added some more (unrelated) test cases and this has been released in [`3.3.5`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v3.3.5) 🚀
